In [58]:
#Import the libraries to be used
import numpy as np
import scipy.stats as stats
from tabulate import tabulate

### Question One
Daily energy intake in kJ was measured for 11 women (Altman, 1991):
5260, 5470, 5640, 6180, 6390, 6515, 6805, 7515, 7515, 8230, 8770.
We wish to investigate whether the women’s energy intake deviates
systematically from a recommended value of 7725 kJ. Assuming this
data comes from a normal distribution; use a t-test to test whether the
distribution might have a mean of 7725 kJ. Explain whether a left-tail,
right-tail or two-tailed test is appropriate. Give the sample mean, sample
standard deviation, standard error of the mean (SEM), t statistic, degrees
of freedom and p-value. Finally explain if the null hypothesis is rejected
or not.

In [66]:
#Load the daily intake data into a list and convert it to a numpy array
daily_intake = np.array([5260, 5470, 5640, 6180, 6390, 6515, 6805, 7515, 7515, 8230, 8770])
#Get the size pf the dataset
sample_size = np.size(daily_intake)

In [73]:
#Set the recommended mean value
recommended_mean = 7725
#Use the scipy ttest_1samp function to perform a one-sample t-test
t_statistic, p_value = stats.ttest_1samp(daily_intake, recommended_mean)

#Compute the mean of the daily intake data
sample_mean = round(np.mean(daily_intake), 2)

#Compute the standard deviation of the daily intake data
sd = round(np.std(daily_intake), 2)

#Compute the standard error of the daily intake data
sem_val = np.std(daily_intake, ddof=1) / np.sqrt(np.size(daily_intake))
sem = round(sem, 2)

#Round off the t-statistic
t_stat = (round(t_statistic, 2))

#Round off the p-value
p_val = (round(p_value, 2))

#Compute the degrees of freedom
df = sample_size - 1

table = [["Mean", sample_mean],["Standard Deviation",sd],["Sample Standard error of mean",sem],["T-Statistic",t_stat],
         ["Degrees of Freedom",df],["P-Value",p_val]]
print(tabulate(table, headers=["Statistic", "Value"]))

Statistic                        Value
-----------------------------  -------
Mean                           6753.64
Standard Deviation             1088.97
Sample Standard error of mean   344.36
T-Statistic                      -2.82
Degrees of Freedom               10
P-Value                           0.02


**Setting the null and alternative hypotheses:**

H0 (Null Hypothesis): μ = 7725 kJ

HA (Alternative Hypothesis): μ ≠ 7725 kJ

**Stating the significance level:**

α = 0.05

**Determining the approriate test type**

A two-tailed test is appropriate because our alternative hypothesis is testing for a value that is not equal to the test statistic

**Conclusion**

The mean of data provided is 6753.64 kJ, which is significantly greater than the sample mean value of 7725 kJ. Therefore the null hypothesis is rejected because μ ≠ 7725 kJ 

**Or**

According to the t-Distribution table, the t critical value at α = 0.05 and n-1 = 10 degrees of freedom is 1.812. This is not less than our t-statistic value of -2.82. In addition, the p-value of the data provided is 0.02, which is less than our significance level of 0.05 therefore the null hypothesis is rejected

### Question Two 
 A Guinness Overall Enjoyment Score (GOES) was used to test if Guinness served in an Irish pub tastes significantly better than pints served elsewhere around the globe. Pints consumed in Ireland received a mean GOES score of 74, while the average GOES score for Guinness tasted elsewhere was 57. The full results were as follows:

    Location Sample Size Mean Standard Deviation
    Ireland 42 74 7.4
    Elsewhere 61 57 7.1
    
Is this difference of 74 versus 57 significant, or is it simply due to natural, random variation? Use a t-test and explain whether a one-sample, two-sample or paired test is appropriate. Show the steps of calculating the t statistic and explain whether a left-tailed, right-tailed or two-tailed test is required. Give the resulting p-value.

**Setting the null and alternative hypotheses:**

H0 (Null Hypothesis): μ = 7725 kJ

HA (Alternative Hypothesis): μ ≠ 7725 kJ

**Stating the significance level:**

α = 0.05

**Determining the approriate test type**

A two-tailed test is appropriate because our alternative hypothesis is testing for a value that is not equal to the test statistic

**Conclusion**

The mean of data provided is 6753.64 kJ, which is significantly greater than the sample mean value of 7725 kJ. Therefore the null hypothesis is rejected because μ ≠ 7725 kJ 

**Or**

According to the t-Distribution table, the t critical value at α = 0.05 and n-1 = 10 degrees of freedom is 1.812. This is not less than our t-statistic value of -2.82. In addition, the p-value of the data provided is 0.02, which is less than our significance level of 0.05 therefore the null hypothesis is rejected